crates/qasm2/src/bytecode.rs

```diff
@@ -211,89 +211,108 @@ pub enum InternalBytecode {
     },
 }
 
-impl IntoPy<Bytecode> for InternalBytecode {
+impl<'py> IntoPyObject<'py> for InternalBytecode {
```

Comment:
Is this not derivable because of the enum variants have named fields?

Reply:
Yeah when I did the derive approach it didn't work, I don't remember the exact details because I worked on this > 1 month ago. But I had to go this route.

Reply:
Using the semantics of the derive macros would substantially change how the Rust/Python boundary layer is handled here.  Even if PyO3 didn't error, the types would be entirely different to what Python is expecting.